lib/spack/spack/solver/asp.py

```diff
@@ -1847,7 +1847,7 @@ def os_defaults(self, specs):
         # Consider any OS's mentioned on the command line. We need this to
         # cross-concretize in CI, and for some tests.
         # TODO: OS should really be more than just a label -- rework this.
-        for spec in specs:
+        for spec in spack.traverse.traverse_nodes(specs):
```

Comment:
Is this specific to this PR, or just an optimization to avoid multiple traversals of input specs?